db/db_impl/db_impl.cc

```diff
@@ -1980,31 +2003,46 @@ Status DBImpl::GetImpl(const ReadOptions& read_options, const Slice& key,
   if (!skip_memtable) {
     // Get value associated with key
     if (get_impl_options.get_value) {
-      if (sv->mem->Get(lkey, get_impl_options.value->GetSelf(), timestamp, &s,
-                       &merge_context, &max_covering_tombstone_seq,
-                       read_options, false /* immutable_memtable */,
-                       get_impl_options.callback,
-                       get_impl_options.is_blob_index)) {
+      if (sv->mem->Get(
+              lkey,
+              get_impl_options.value ? get_impl_options.value->GetSelf()
```

Comment:
For my understanding, if `get_impl_options.get_value` is true can `get_impl_options.value` be `nullptr`?

Reply:
Yes, that's the case with the new `GetEntity` API. So for `Get`, we have `get_value == true && value != nullptr && columns == nullptr`; for `GetEntity`, we have `get_value == true && value == nullptr && columns != nullptr`.